opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -52,37 +56,54 @@
    *
    * @param <C> the carrier type, which also parametrizes the Format.
    * @param spanContext the SpanContext instance to inject into the carrier
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.inject() implementations must support io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.inject() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
+   *
+   * @throws io.opentracing.propagation.UnsupportedFormatException when tracer does not support given format
    *
    * @see io.opentracing.propagation.Format
    * @see io.opentracing.propagation.Format.Builtin
+   * @see io.opentracing.propagation.UnsupportedFormatException
    */
   <C> void inject(SpanContext spanContext, Format<C> format, C carrier);
 
   /**
    * Extract a SpanContext from a `carrier` of a given type, presumably after propagation across a process boundary.
    *
+   * If tracer does not support given format, it must throw
+   * io.opentracing.propagation.UnsupportedFormatException so that instrumentation
+   * can try another format.
+   *
+   * If given carrier does not contain information to construct SpanContext,
+   * the tracer must return `null`. If SpanContext cannot be extracted because
+   * the data in the carrier is malformed, the tracer should return null, rather
+   * than throw an exception. The tracer may handle the exceptions internally
+   * by logging them or incrementing an error counter.
+   *
    * <p>Example:
    * <pre>{@code
    * Tracer tracer = ...
    * TextMap httpHeadersCarrier = new AnHttpHeaderCarrier(httpRequest);
    * SpanContext spanCtx = tracer.extract(Format.Builtin.HTTP_HEADERS, httpHeadersCarrier);
-   * tracer.buildSpan('...').withChildOf(spanCtx).start();
+   * tracer.buildSpan('...').asChildOf(spanCtx).start();
    * }</pre>
    *
    * If the span serialized state is invalid (corrupt, wrong version, etc) inside the carrier this will result in an
    * IllegalArgumentException.
    *
    * @param <C> the carrier type, which also parametrizes the Format.
-   * @param format the Format of the carrier
-   * @param carrier the carrier for the SpanContext state. All Tracer.extract() implementations must support
-   *                io.opentracing.propagation.TextMap and java.nio.ByteBuffer.
+   * @param format the Format of the carrier. All Tracer.extract() implementations
+   *               must support formats defined in io.opentracing.propagation.Format.Builtin
+   * @param carrier the carrier for the SpanContext state.
    *
-   * @return the SpanContext instance holding context to create a Span.
+   * @return the SpanContext instance extracted from the carrier holding context to create a Span or null if it cannot be extracted.
```

Comment:
I am not in favor of making an API that lies to the users. Knowing if there was or wasn't a tracing context in the incoming request may be important to some consumers. By returning an empty context we're loosing that bit of information, because the context interface has no properties to identify itself as "empty".